address_book.rb

```diff
@@ -0,0 +1,293 @@
+         puts "Enter Street"
+         street=gets.chomp
+       end
+       puts"city"
+       city=gets.chomp
+       while !( isAlpha(city)) do
+         puts "Invalid city"
+         puts "Enter city"
+         city=gets.chomp
+       end
+       puts "state"
+       state=gets.chomp
+       while !( isAlpha(state)) do
+         puts "Invalid state"
+         puts "Enter state"
+         state=gets.chomp
+       end
+       puts "zip "
+       zip=gets.chomp
+       while !( isNum(zip)) do
+         puts "Invalid Zip "
+         puts "Enter Zip"
+         zip=gets.chomp
+       end
+   end
+       addr = Address.new(street,city,state,zip)
+       @@contacts.push(Person.new(fname,lname,pnum,addr))
+
+       end
+
+
+
+
+   def displayContactByIndex(i)
+
+    puts @@contacts[i].getfirstName + " " + @@contacts[i].getlastName + ":"
+    puts "       #{@@contacts[i].getphoneNum}"
+    puts "       #{@@contacts[i].getAddress.getStreet}"
+    puts "       #{@@contacts[i].getAddress.getCity}"
+    puts "       #{@@contacts[i].getAddress.getState} , #{@@contacts[i].getAddress.getZip}"
+
+   end
+
+    def displayAllContacts
+
+      for i in 0..@@contacts.length-1
+        displayContactByIndex(i)
+      end
+    end
+
+  def searchContact
+    puts "Enter name to Search details"
+    fname=gets.chomp
+    counts=occurance(fname)
+    #print counts
+    if(counts==0)
+      puts "No Such Contact"
+    else
+     for i in 0..@@contacts.length-1
+       if fname==@@contacts[i].getfirstName
+        displayContactByIndex(i)
+       end
+     end
+    end
+  end
+
+  def sortContacts
+
+    @@contacts.sort_by!{ |c| c.getfirstName+c.getlastName.downcase }
+    puts "After Sorting"
+    displayAllContacts
+
+  end
+
+
+  def deleteContact
+    puts "Enter Contact name to Delete"
+    fname = gets.chomp
+     counts=occurance(fname)
+
+      if(counts==0)
+        puts "No Such Contact"
+      elsif(counts>1)
+        puts "Enter last Name"
+        lname=gets.chomp
+        isExists=false
+        for i in 0..@@contacts.length-1
+          if fname==@@contacts[i].getfirstName and lname==@@contacts[i].getlastName
+            isExists=true
+            puts "Are you sure? You want to delete #{fname}? Y/N"
+            delete=gets.chomp
+            if(delete='y'||delete='Y')
+              puts "#{@@contacts[i].getfirstName} is deleted Successfully"
+              @@contacts.delete(@@contacts[i])
+
+            end
+            break
+          end
+        end
+        if(isExists==false)
+          puts "No such Contact"
+        end
+      else
+        for i in 0..@@contacts.length-1
+          if fname==@@contacts[i].getfirstName
+            puts "Are you sure? You want to delete #{fname}? Y/N"
+            delete=gets.chomp
+            if(delete='y'||delete='Y')
+            puts "#{@@contacts[i].getfirstName} is deleted Successfully"
+            @@contacts.delete(@@contacts[i])
+            end
+            break
+          end
```

Comment:
The code in the else condition can be included can be avoided and generalized for both accepting first name and last name, if there are more than 1 contacts present.

Reply:
I dont see the code change onthis comment. Please work on this